packages/next-codemod/bin/upgrade.ts

```diff
@@ -651,3 +661,119 @@ function writeOverridesField(
     }
   }
 }
+
+function warnPeerDependencies(
+  appPackageJson: any,
+  versionMapping: Record<string, { version: string; required: boolean }>
+) {
+  const versionMappingKeys = Object.keys(versionMapping)
+  const allDependenciesToInstall = {
+    ...appPackageJson.dependencies,
+    ...appPackageJson.devDependencies,
+  }
+  const peerDepsOutOfRange = new Map<
+    string,
+    { current: string; expected: string; from: string }
+  >()
+
+  if ('peerDependencies' in appPackageJson) {
+    const peerDeps = appPackageJson.peerDependencies
+    const peerDepsKeys = Object.keys(peerDeps)
+    const commonKeys = versionMappingKeys.filter((versionMappingKey) =>
+      peerDepsKeys.includes(versionMappingKey)
+    )
+
+    if (commonKeys.length > 0) {
+      for (const key of commonKeys) {
+        const peerDepVersion = peerDeps[key]
+        const { version } = versionMapping[key]
+
+        if (
+          !semverSatisfies(version, peerDepVersion, {
+            includePrerelease: true,
+          })
+        ) {
+          peerDepsOutOfRange.set(key, {
+            current: version,
+            expected: peerDepVersion,
+            from: key,
+          })
+        }
+      }
+    }
+  }
+
+  for (const dependency of Object.keys(allDependenciesToInstall)) {
+    // use readFile to ensure it reads after the install is complete
+    const pkgJson = JSON.parse(
+      fs.readFileSync(
+        path.join(cwd, 'node_modules', dependency, 'package.json'),
+        'utf8'
+      )
+    )
+
+    if ('peerDependencies' in pkgJson) {
+      const peerDeps = pkgJson.peerDependencies
+      const peerDepsKeys = Object.keys(peerDeps)
+      const commonKeys = versionMappingKeys.filter((versionMappingKey) =>
+        peerDepsKeys.includes(versionMappingKey)
+      )
+
+      if (commonKeys.length > 0) {
+        for (const key of commonKeys) {
+          const peerDepVersion = peerDeps[key]
+          const { version } = versionMapping[key]
+
+          if (
+            !semverSatisfies(version, peerDepVersion, {
+              includePrerelease: true,
+            })
+          ) {
+            peerDepsOutOfRange.set(key, {
+              current: version,
+              expected: peerDepVersion,
+              from: dependency,
+            })
+          }
+        }
+      }
+    }
+  }
+
+  const size = peerDepsOutOfRange.size
+  if (size > 0) {
+    console.log(
+      `${pc.yellow('⚠')} Found ${size} peer ${
+        size === 1 ? 'dependency' : 'dependencies'
+      } out of range from the direct dependencies.`
```

Comment:
Need to polish this. The wording is a bit off here. Don't have a suggestion just yet though.

Reply:
How about this?

```ts
console.log(
  `${pc.yellow('⚠')} Found ${size} ${
    size === 1 ? 'dependency' : 'dependencies'
  } unmet peer dependency range with the updated packages.`
)
```

Reply:
@eps1lon Need review as well!